client/src/components/App.jsx

```diff
@@ -12,15 +14,21 @@ max-width: 1500px;
 margin: auto;
 `
 
+export async function productLoader({ params }) {
+  const product = await api.getProductId(params.productId);
+  return { product };
+};
+
```

Comment:
I would love to get a quick rundown tomorrow on how this exported async function gets used. Intruiging!